src/nvim/memory.c

```diff
@@ -92,6 +100,12 @@ void *xmalloc(size_t size)
   return ret;
 }
 
+/// free wrapper that returns delegates to the backing memory manager
+void xfree(void *ptr)
+{
+  free(ptr);
```

Comment:
In this module, `free` is only a placeholder for the actual `free` implementation(which is `je_free` when `USE_JEMALLOC` is defined). We still need a fixed function address for the other modules to link against. Redefining `free` as a macro globally is definitely not a good approach(we lose sanity when setting breakpoints, for example)

> Is the plan to call je_* explicitly from the xfoo family, in the future?

No, but we could eventually use this as a slot for extra deallocation logic. For example, see how redis keeps track of how much memory is currently used in [zmalloc.c](https://github.com/antirez/redis/blob/unstable/src/zmalloc.c).